app/src/main/java/com/example/mapsandcollections/dto/MapItemModel.java

```diff
@@ -0,0 +1,166 @@
+package com.example.mapsandcollections.dto;
+
+import android.content.Context;
+import android.os.Handler;
+import android.os.Looper;
+
+import com.example.mapsandcollections.R;
+import com.example.mapsandcollections.components.tasker.Tasker;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.TreeMap;
+import java.util.concurrent.Callable;
+
+public class MapItemModel implements IItemTaskModel {
+
+    private String HASH_MAP;
+    private String TREE_MAP;
+
+    private String ADD_NEW;
+    private String REMOVE;
+    private String SEARCH_BY_KEY;
+
+    private final Handler handler;
+    private final Context context;
+    private List<ItemTask> itemTaskList;
+    private Map<Object, Object> hashMap;
+    private Map<Object, Object> treeMap;
+    private Tasker.OnTaskDoneListener listener;
+
+    MapItemModel(Context context) {
+        this.handler = new Handler(Looper.getMainLooper());
+        this.context = context;
+        initStrings();
+        createModel();
+    }
+
+    private void initStrings() {
+        HASH_MAP = context.getString(R.string.hash_map);
+        TREE_MAP = context.getString(R.string.tree_map);
+
+        ADD_NEW = context.getString(R.string.add_new);
+        REMOVE = context.getString(R.string.remove);
+        SEARCH_BY_KEY = context.getString(R.string.search_by_key);
+    }
+
+    private void createModel() {
+
+        final Random random = new Random();
+        itemTaskList = new ArrayList<>();
+        itemTaskList.add(new ItemTask(0, HASH_MAP, ADD_NEW, () -> {
```

Comment:
tree map and hash map implement interface Map<KEY, VAL>
think how to reduce an amount of hardcoded tasks